src/Features/CSharp/Portable/GenerateMember/GenerateVariable/CSharpGenerateVariableService.cs

```diff
@@ -33,7 +32,17 @@ protected override bool IsExplicitInterfaceGeneration(SyntaxNode node)
             => node is PropertyDeclarationSyntax;
 
         protected override bool IsIdentifierNameGeneration(SyntaxNode node)
-            => node is IdentifierNameSyntax identifier && !identifier.Identifier.CouldBeKeyword();
+            => node is IdentifierNameSyntax identifierName && !IsProbablySyntacticConstruct(identifierName.Identifier);
+
+        private static bool IsProbablySyntacticConstruct(SyntaxToken token)
+        {
+            var contextualKind = SyntaxFacts.GetContextualKeywordKind(token.ValueText);
+
+            return contextualKind is SyntaxKind.FromKeyword or
+                                  SyntaxKind.NameOfKeyword or
+                                  SyntaxKind.AsyncKeyword or
+                                  SyntaxKind.AwaitKeyword;
```

Comment:
'var'?

Reply:
Why `var`? `nameof` is probably `nameof(something)`, `async` can start delegate declaration, `await` is obvious). What construct can start `var`? `return var what?`, `Foo(var what?)`

Reply:
'var' is probably a syntactic construct.  Almost all users would be using it to start writing `var x`.